numpy/core/src/multiarray/ctors.c

```diff
@@ -2024,7 +2024,7 @@ PyArray_FromArray(PyArrayObject *arr, PyArray_Descr *newtype, int flags)
         newtype = oldtype;
         Py_INCREF(oldtype);
     }
-    if (PyDataType_ISUNSIZED(newtype)) {
+    else if (PyDataType_ISUNSIZED(newtype)) {
```

Comment:
This bug manifested itself as `typestr="|V0"` causing the resulting array to be a copy rather than a view